indexer/scripts/elastic-conf.py

```diff
@@ -96,6 +116,50 @@ def main_loop(self) -> None:
             logger.error("One or more configurations failed. Check logs for details.")
             return
 
+    def repository_exists(self, es: Elasticsearch, repo_name: str) -> bool:
+        status = False
+        try:
+            response = es.snapshot.get_repository(name=repo_name)
+            if repo_name in response:
+                status = True
+        except Exception as e:
+            logger.error("Error validating repository: %s", e)
+        return status
```

Comment:
```suggestion
    def _repository_exists(self, es: Elasticsearch) -> bool:
        try:
            response = es.snapshot.get_repository(name=self.es_snapshot_repo)
            return repo_name in response
        except Exception as e:
            logger.exception("Error validating repository: %s", name)
        return False
```

1. repo name should be available to all methods in this class (self.`es_snapshot_repo`)
2. [`logger.exception`](https://docs.python.org/3/library/logging.html#logging.Logger.exception) includes error stack trace by default (if you wanted to include `e` in log message)